airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,148 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
                     {properties}"""
             )
 
+    def test_property_type_is_not_array(self, connector_spec_dict: dict):
+        """
+        Each field has one or multiple types, but the UI only supports a single type and optionally "null" as a second type.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, value in dpath.util.search(specification, "**/properties/*/type", yielded=True):
+            if isinstance(value, List):
+                if len(value) != 2:
+                    errors.append(f"{type_path} is not either a simple type or an array of a simple type plus null: {value}")
+                if value[1] != "null":
+                    errors.append(
+                        f"Second type of {type_path} is not null: {value}. Type can either be a simple type or an array of a simple type plus null"
+                    )
+        if len(errors) > 0:
+            pytest.fail("\n".join(errors))
+
+    @staticmethod
+    def _get_parent(obj: dict, path: str) -> Any:
+        """
+        Returns the parent dict of a given path within the `obj` dict
+        """
+        absolute_path = f"/{path}"
+        parent_path, _ = absolute_path.rsplit(sep="/", maxsplit=1)
+        if parent_path == "":
+            return obj
+        return dpath.util.get(obj, parent_path)
+
+    def test_object_not_empty(self, connector_spec_dict: dict):
+        """
+        Each object field needs to have at least one property as the UI won't be able to show them otherwise.
+        If the whole spec is empty, it's allowed to have a single empty object at the top level
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, value in dpath.util.search(specification, "**/type", yielded=True):
+            if type_path == "type":
+                # allow empty root object
+                continue
+            if value == "object":
+                property = self._get_parent(specification, type_path)
+                if "oneOf" not in property and ("properties" not in property or len(property["properties"]) == 0):
+                    errors.append(
+                        f"{type_path} is an empty object which will not be represented correctly in the UI. Either remove or add specific properties"
+                    )
+        if len(errors) > 0:
+            pytest.fail("\n".join(errors))
+
+    def test_array_type(self, connector_spec_dict: dict):
+        """
+        Each array has one or multiple types for its items, but the UI only supports a single type which can either be object, string or an enum
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        errors = []
+        for type_path, type in dpath.util.search(specification, "**/type", yielded=True):
```

Comment:
```suggestion
        for type_path, type_value in dpath.util.search(specification, "**/type", yielded=True):
```
For consistency, moreover `type` is a reserved python keyword.